pkg/storage/factory_test.go

```diff
@@ -45,6 +50,60 @@ func TestFactoryStop(t *testing.T) {
 	store.Stop()
 }
 
+func TestNamedStores(t *testing.T) {
```

Comment:
Maybe I missed it, but I was expecting to find a test that shows that named and non-named stores can be used together?
Something like "period A is using the default GCS bucket, but period B is using a named store with a different bucket also in GCS"

Reply:
problem is when using boltdb or tsdb indexes, current code only intializes a single instance based on `sharedStoreType` or the objectType from the latest period config.

Next [PR](https://github.com/grafana/loki/pull/7754) will remove this limitation and adds tests with two stores [1](https://github.com/grafana/loki/pull/7754/files#diff-d383bd6a92c29d4c84d56fdd2ac42de2991fd4c2c38eb0a380e1c5d8099fa946) [2](https://github.com/grafana/loki/pull/7754/files#diff-556b2c27e0ba1456d41699b8a0f631c10bce22f01d15b860f87a8ce9b3a7ef16)

Reply:
OK cool :+1: